packages/fxa-payments-server/src/lib/flow-event.ts

```diff
@@ -0,0 +1,51 @@
+interface FlowEventData {
+  deviceId: string;
+  flowBeginTime: number;
+  flowId: string;
+}
+
+let initialized = false;
+let optEventData: FlowEventData;
+
+function buildAmplitudeTypeName(groupName: string, eventName: string) {
+  return `amplitude.${groupName}.${eventName}`;
+}
+
+export function init(eventData: any = {}) {
+  if (!initialized) {
+    optEventData = {
+      deviceId: eventData.device_id,
+      flowBeginTime: eventData.flow_begin_time,
+      flowId: eventData.flow_id,
```

Comment:
If these flow params should always be present, I think we should change the type signature on this function to make `eventData` specify that the keys need to be there and should at least be strings. If the flow params might be missing, we should figure out what to do to avoid propagating `undefined` values here.

Reply:
I don't know which of the flow params must be present.  (Maybe need to look in fxa-amplitude-send?)  But yeah, I won't set `initialized` to true unless all the flow params are not empty.